core/src/Kokkos_Macros.hpp

```diff
@@ -369,10 +373,14 @@
 #define KOKKOS_FORCEINLINE_FUNCTION \
   KOKKOS_IMPL_FORCEINLINE_FUNCTION  \
   __attribute__((annotate("KOKKOS_FORCEINLINE_FUNCTION")))
+#define KOKKOS_EXTERNAL_FUNCTION \
+  KOKKOS_IMPL_EXTERNAL_FUNCTION  \
+  __attribute__((annotate("KOKKOS_EXTERNAL_FUNCTION")))
```

Comment:
Why are you introducing a new macro?

Reply:
SYCL needs SYCL_EXTERNAL for functions used with RDC/separable compilation (https://www.intel.com/content/www/us/en/developer/articles/technical/use-the-sycl-external-macro.html, https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:syclexternal) but doesn't need any annotations for "normal" device functions in contrast to CUDA/HIP so the idea is to add a portability macro.